tasks/main.yml

```diff
@@ -68,11 +68,26 @@
   when: docker_daemon_options.keys() | length > 0
   notify: restart docker
 
+- name: Replace Docker service ExecStart command.
+  when: docker_service_start_command != ""
+  notify: restart docker
+  block:
+    - name: Get Docker service status
+      ansible.builtin.systemd_service:
+        name: docker
+      register: docker_service_status
+    - name: Patch docker.service
+      ansible.builtin.replace:
+        path: "{{ docker_service_status.status['FragmentPath'] }}"
+        regexp: "^ExecStart=.*$"
+        replace: "ExecStart={{ docker_service_start_command }}"
+
 - name: Ensure Docker is started and enabled at boot.
   service:
     name: docker
     state: "{{ docker_service_state }}"
     enabled: "{{ docker_service_enabled }}"
+    daemon_reload: true
```

Comment:
Would this cause the daemon to be reloaded on every run?

Reply:
Yes. I just looked into the documentation. I will look for a better solution :smile: 

---
https://docs.ansible.com/ansible/latest/collections/ansible/builtin/systemd_service_module.html#parameter-daemon_reload